src/main/java/net/openhft/chronicle/bytes/PageUtil.java

```diff
@@ -0,0 +1,146 @@
+import net.openhft.chronicle.core.OS;
+import net.openhft.chronicle.core.annotation.Positive;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.text.MessageFormat.format;
+
+/**
+ * Provides file systems' page size auto-resolving.
+ * Linux and hugetlbfs specific only for now.
+ */
+public final class PageUtil {
+
+    public static final int DEFAULT_HUGE_PAGE_SIZE = 2 * 1024 * 1024;
+
+    private static final Pattern PAGE_SIZE_PATTERN = Pattern.compile("pagesize=([0-9])+([KkMmGg])");
+    private static final TrieNode root = new TrieNode();
+
+    static {
+        if (OS.isLinux()) {
+            List<String> mounts = readMountInfo("/proc/self/mountinfo");
+            for (String mount : mounts) {
+                if (mount.contains("hugetlbfs")) {
+                    int size = parsePageSize(mount);
+                    String path = parseMountPoint(mount);
+                    insert(path, size);
+                }
+            }
+        }
+    }
+
+    private static List<String> readMountInfo(String path) {
+        try {
+            return Files.readAllLines(Paths.get(path));
+        } catch (IOException e) {
+            Jvm.warn().on(PageUtil.class, format("Error reading ''{0}'': {1}", path, e.getMessage()));
+            return Collections.emptyList();
+        }
+    }
+
+    private static int parsePageSize(String mount) {
+        Matcher matcher = PAGE_SIZE_PATTERN.matcher(mount);
+        if (matcher.find())
+            try {
+                return Integer.parseInt(matcher.group(1)) * mult(matcher.group(2));
+            }
+        catch (Exception e) {
+            Jvm.warn().on(PageUtil.class, format("Error parsing pageSize={0}: {1}", matcher.group(1), e.getMessage()));
+        }
+        return DEFAULT_HUGE_PAGE_SIZE;
+    }
+
+    private static int mult(String s) {
+        int k = 1024;
+        if (s.equalsIgnoreCase("K"))
+            return k;
+        if (s.equalsIgnoreCase("G"))
+            return k * 1024 * 1024;
+
+        return k * 1024;
+    }
+
+    private static String parseMountPoint(String line) {
+        String[] parts = line.split("\\s+");
+        assert parts.length == 11;
+        assert parts[4].matches("^\\S+$");
+        return parts[4];
+    }
+
+    private static void insert(String path, int size) {
+        TrieNode curr = root;
+        for (String dir : path.split("/")) {
+            if (dir.isEmpty()) continue;
+            curr.childs.putIfAbsent(dir, new TrieNode());
+            curr = curr.childs.get(dir);
+        }
+        curr.isLeaf = true;
+        curr.pageSize = size;
+        Jvm.warn().on(PageUtil.class, format("Found pageSize={0} for mount point ''{1}''", size, path));
```

Comment:
Please could we change this to info level. Its presence in some Queue tests causes them to fail as the tests are configured to fail on warnings (we could ignore them if preferred).

I don't believe there is a Jvm.info() though and it might be outside of the scope of this change to add it. Let me know what you think

Reply:
Changed to info (apparently it's called PERF).